x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java

```diff
@@ -369,6 +371,7 @@ private void handleFailure(Exception e, AtomicInteger retryCounter, Runnable tas
             scheduler.accept(TimeValue.timeValueMillis(delay), task);
         } else {
             fatalException = ExceptionsHelper.convertToElastic(e);
+            LOGGER.warn("shard follow task encounter non-retryable error", e);
```

Comment:
👍

Reply:
@martijnvg Should we also mark the follow-task as failed in this case?

Reply:
Good point.